src/frontend/src/btc/schedulers/btc-wallet.scheduler.ts

```diff
@@ -46,100 +65,159 @@ export class BtcWalletScheduler implements Scheduler<PostMessageDataRequestBtc>
+			data: await getBtcBalance({
+				identity,
+				network: mapToSignerBitcoinNetwork({
+					network: bitcoinNetwork
+				}),
+				minConfirmations: BTC_BALANCE_MIN_CONFIRMATIONS
+			}),
 			certified: true
 		};
+	};
 
-		this.store = {
-			...this.store,
-			balance: certifiedBalance
-		};
+	private loadWalletData = async ({
+		certified,
+		identity,
+		bitcoinNetwork,
+		btcAddress,
+		minterCanisterId,
+		shouldFetchTransactions
+	}: LoadBtcWalletParams) => {
+		const balance = await this.loadBtcBalance({
+			identity,
+			bitcoinNetwork,
+			certified,
+			btcAddress,
+			minterCanisterId
+		});
 
-		return certifiedBalance;
-	}
+		// TODO: investigate and implement "update" call for BTC transactions
+		const transactions =
+			shouldFetchTransactions && !certified
+				? await this.loadBtcTransactionsData({ btcAddress })
+				: [];
+
+		return { balance, transactions };
+	};
 
-	/* TODO: The following steps need to be done:
-	 * 1. [Required] Fetch uncertified transactions via BTC transaction API.
-	 * 2. [Improvement] Query uncertified balance in oder to improve UX (signer.getBtcBalance takes ~5s to complete).
-	 * 3. [Required] Fetch certified transactions via BE endpoint (to be discussed).
-	 * */
 	private syncWallet = async ({ identity, data }: SchedulerJobData<PostMessageDataRequestBtc>) => {
 		const bitcoinNetwork = data?.bitcoinNetwork;
-		assertNonNullish(bitcoinNetwork, 'No BTC network provided to get BTC certified balance.');
+		assertNonNullish(bitcoinNetwork, 'No BTC network provided to get BTC balance.');
 
 		const btcAddress = data?.btcAddress.data;
 		assertNonNullish(btcAddress, 'No BTC address provided to get BTC transactions.');
 
-		const balance = await this.loadBtcBalance({ identity, bitcoinNetwork });
-		const { newTransactions, latestBitcoinBlockHeight } =
-			nonNullish(data) && data.shouldFetchTransactions
-				? await this.loadBtcTransactionsData({ btcAddress })
-				: { newTransactions: [], latestBitcoinBlockHeight: undefined };
+		// TODO: implement "onCertifiedError" to handle errors in update calls
+		await queryAndUpdate<BtcWalletData>({
+			request: ({ identity: _, certified }) =>
+				this.loadWalletData({
+					certified,
+					identity,
+					btcAddress,
+					bitcoinNetwork,
+					shouldFetchTransactions: data?.shouldFetchTransactions,
+					minterCanisterId: data?.minterCanisterId
+				}),
+			onLoad: ({ certified, ...rest }) => this.syncWalletData({ certified, ...rest }),
+			identity,
+			resolution: 'all_settled'
+		});
+	};
 
-		// TODO: handle the case when tx data is available but latestBitcoinBlockHeight is undefined
-		const uncertifiedTransactions = nonNullish(latestBitcoinBlockHeight)
-			? newTransactions.map((transaction) => ({
-					data: mapBtcTransaction({ transaction, btcAddress, latestBitcoinBlockHeight }),
-					certified: false
-				}))
-			: [];
+	private syncWalletData = ({
+		response: { balance, transactions }
+	}: {
+		response: BtcWalletData;
+		certified: boolean;
```

Comment:
Should we remove/exclude the `certified` parameter? I spent quite some time reviewing this so it would maybe be useful to make it clear we are considering another certified information/source.

Reply:
Removed.